src/mixins/segmentation/freehandSegmentationMixin.js

```diff
@@ -0,0 +1,189 @@
+import external from '../../externalModules.js';
+import store from '../../store';
+import { getLogger } from '../../util/logger.js';
+import { draw, drawJoinedLines, getNewContext } from '../../drawing';
+
+const logger = getLogger('tools:ScissorsTool');
+
+const { getters, setters } = store.modules.segmentation;
+
+/**
+ * Render hook: draws the FreehandScissors's outline
+ *
+ * @param {Object} evt Cornerstone.event#cornerstoneimagerendered > cornerstoneimagerendered event
+ * @memberof Tools.ScissorsTool
+ * @returns {void}
+ */
+function renderToolData(evt) {
+  const eventData = evt.detail;
+  const { element } = eventData;
+  const color = getters.brushColor(element, true);
+  const context = getNewContext(eventData.canvasContext.canvas);
+  const handles = this.handles;
+
+  draw(context, context => {
+    if (handles.points.length > 1) {
+      for (let j = 0; j < handles.points.length; j++) {
+        const lines = [...handles.points[j].lines];
+        const points = handles.points;
+
+        if (j === points.length - 1) {
+          // If it's still being actively drawn, keep the last line to
+          // The mouse location
+          lines.push(this.handles.points[0]);
+        }
+        drawJoinedLines(context, element, this.handles.points[j], lines, {
+          color,
+        });
+      }
+    }
+  });
+}
+
+/**
+ * Sets the start handle point and claims the eventDispatcher event
+ *
+ * @private
+ * @param {*} evt // mousedown, touchstart, click
+ * @returns {void|null}
+ */
+function _startOutliningRegion(evt) {
```

Comment:
Is it weird that our mixin adds a private method that's not used by other mixin methods? Is the assumption that this mixin can only be added to a `BaseBrushTool` or the now dead `BaseSegmentationTool`? If so, how do we constrain/inform for that?

Reply:
The mixin makes the tool a segmentation tool, with its own input loop defined by the segmentation mixin, and strategies implemented by the tool itself. I think we can make this clear just with good documentation, and with segmentation mixins living in one centralised location.